streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/PublishSubjectImpl.kt

```diff
@@ -75,26 +81,22 @@ open class PublishSubjectImpl<T> : PublishSubject<T> {
     }
 
     private fun addSubscription(subscription: PublisherSubscription<T>) {
+        if (!subscription.isCancelled) {
+            onNewSubscription(subscription)
             if (!subscription.isCancelled) {
-                onNewSubscription(subscription)
-                if (!subscription.isCancelled) {
-                    serialQueue.dispatch {
-                        if (this.completed) {
-                            subscription.dispatchCompleted()
-                        } else if (subscriptions.add(subscription).count() == 1) {
-                            onFirstSubscription()
-                        }
-                    }
+                if (this.completed) {
+                    subscription.dispatchCompleted()
+                } else if (subscriptions.add(subscription).count() == 1) {
+                    onFirstSubscription()
                 }
             }
+        }
     }
 
     protected fun cleanupValues() {
         if (hasSubscriptions) throw IllegalStateException("Cannot clean values when publisher has subscribers")
-        serialQueue.dispatch {
```

Comment:
Why should we not dispatch these values on the serialQueue anymore?

Reply:
It caused problems in the resubscription phase since values where cleaned up too late. When the ColdPublisher was reconnected, the previous error was re-emitted because of that.

I talked with @martingagnon and he was against sharing dispatch queue with parent and dispatching the reconnection on serial queue.

Reply:
It does not break any unit tests so my guess is that it was not required to be serial otherwise there would be a test for it.

Reply:
Allright 👍